### ISSUE TITLE
Source code of the models

### ISSUE THREAD

**Issue by hadim** [agemagician/ProtTrans#11]
Do you have the source code of the various pre-trained models?

**Comment by agemagician** [comment]
We have used the official training scripts for each model, here is the list of the training source code for each model:

- Bert: https://github.com/google-research/bert
- Albert: https://github.com/google-research/albert
- Electra: https://github.com/google-research/electra
- T5: https://github.com/google-research/text-to-text-transfer-transformer
- XLNet: https://github.com/zihangdai/xlnet
- Transformer XL: https://github.com/cybertronai/transformer-xl

**Comment by hadim** [comment]
Thank you @agemagician for the answer. Do you also have the preprocessing code as well as the hyperparameters used for all the models? I know some of those are in the paper but having access to the source code would help for reproducibility purposes.

**Comment by agemagician** [comment]
We have used the official training scripts preprocessing parameters and models hyperparameters except if otherwise mentioned on the paper.

**Comment by hadim** [comment]
Unless I miss something, tokenization must be different than for regular NLP models. 

Why don't you want to give access to the original source code used to train the models?

**Comment by agemagician** [comment]
The tokenisation steps were explained on the paper in section "2.1 Data for Language Models (LMs)", which is same as NLP:
`Both UniRef100 and BFD were tokenized with a single space (indicating word-boundaries) between each token. Each protein sequence was stored on a separate line, with lines/proteins representing the equivalent of “sentences”. Additionally, an empty line was inserted between each protein sequence in order to indicate the “end of a document” as some LMs such as Bert use consecutive sequences for an auxiliary task, i.e. next-sentence prediction, which was not used in this work.`

The only minor filtering step was also mentioned on the paper on section "2.1 Data for Language Models (LMs)":
`As a minor filtering step, all non-generic or unresolved amino acids (B, O, U, Z) were mapped to ‘unknown’ (X).`

I already pointed to you on the first reply that we have used the official training scripts, with links to all of them.

I am trying to help but I am not sure what is not clear here.
